SF iOS/SF iOS/Feed/EventsFeedViewController.m

```diff
@@ -201,19 +199,21 @@ - (void)viewDidLoad {
 
     [self registerForPreviewingWithDelegate:self sourceView:self.tableView];
     
-    CGRect searchBarRect = CGRectMake(kSEARCHBARMARGIN,
-                                      0,
-                                      self.view.frame.size.width-(kSEARCHBARMARGIN*2),
-                                      kSEARCHBARHEIGHT);
-    [self setupNotificationsButton];
-    self.searchBar = [[UISearchBar alloc] initWithFrame:searchBarRect];
-    self.searchBar.searchBarStyle = UISearchBarStyleMinimal;
-    self.searchBar.placeholder = NSLocalizedString(@"Filter", @"Prompt to search for event names. Here, `Filter` is a joke in English because people filter coffee and this list can be filtered by a term.");
-    self.searchBar.delegate = self;
-
-    self.tableView.tableHeaderView = self.searchBar;
-    self.tableView.tableHeaderView.backgroundColor = UIColor.whiteColor;
-
+    self.searchController = [[UISearchController alloc] initWithSearchResultsController:nil];
+    self.searchController.searchResultsUpdater = self;
+    self.searchController.searchBar.placeholder = @"Filter";
+    [self.searchController.searchBar sizeToFit];
+    self.searchController.delegate = self;
+    self.searchController.dimsBackgroundDuringPresentation = NO;
+    self.searchController.searchBar.delegate = self;
+    self.navigationItem.searchController = self.searchController;
+    self.navigationItem.hidesSearchBarWhenScrolling = YES;
+    self.definesPresentationContext = YES;
+    self.searchController.hidesNavigationBarDuringPresentation = NO;
```

Comment:
I’ve been putting mega line initializer things like this in their own class. Or at least trying to.